logger/level_test.go

```diff
@@ -0,0 +1,71 @@
+package logger
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestUnmarshalText(t *testing.T) {
+	invalidCases := map[string]struct {
+		input  string
+		output Level
+	}{
+		"select log level Not_A_Level": {"Not_A_Level", 0},
+		"select log level Bad_Input":   {"Bad_Input", 0},
+	}
+
+	validCases := map[string]struct {
+		input  string
+		output Level
+	}{
+		"select log level debug": {"debug", Debug},
+		"select log level DEBUG": {"DEBUG", Debug},
+		"select log level info":  {"info", Info},
+		"select log level INFO":  {"INFO", Info},
+		"select log level warn":  {"warn", Warn},
+		"select log level WARN":  {"WARN", Warn},
+		"select log level Error": {"Error", Error},
+		"select log level ERROR": {"ERROR", Error},
+	}
```

Comment:
Merge these test cases into one map and use one `for` loop.

Reply:
All cleaned up - and thanks again for the tip on adding err to the cases!